File: app/src/main/java/com/botijasoftware/utils/ArrayBufferByte.java
package com.botijasoftware.utils;

import java.nio.ByteBuffer;
import java.nio.ByteOrder;

public class ArrayBufferByte extends ArrayBuffer {
	private ByteBuffer mBuffer;

	public ArrayBufferByte(VertexBufferDefinition vbdefinition, int maxelements) {
		super(vbdefinition, maxelements);

		mBuffer = ByteBuffer.allocateDirect(
				vbdefinition.getDataSize() * vbdefinition.mSize * maxelements)
				.order(ByteOrder.nativeOrder());
	}
	
	public ByteBuffer getBuffer() {
		return mBuffer;
	}
	
	public void put(int index, byte value) {
		mBuffer.put(index, value);
	}
	
	public void put(byte[] data, int offset) {
		mBuffer.put(data, offset, data.length - offset);
		mBuffer.position(0);
	}
	
	public void put(byte[] data) {
		mBuffer.put(data, 0, data.length);
		mBuffer.position(0);
	}
	
	public void put(byte value) {
		mBuffer.put(value);
	}
	
	public void put(short value) {
		mBuffer.put((byte) value);
	}
	
	public void put(int value) {
		mBuffer.put((byte) value);
	}
	
	public void put(float value) {
		mBuffer.put((byte)value);
	}
	
	public void position(int i) {
		mBuffer.position(i);
	}
	
	public int position() {
		return mBuffer.position();
	}

}

File: app/src/main/java/com/botijasoftware/utils/ArrayBufferByte.kt
package com.botijasoftware.utils

import java.nio.ByteBuffer
import java.nio.ByteOrder

class ArrayBufferByte(vbdefinition: VertexBufferDefinition, maxelements: Int) : ArrayBuffer(vbdefinition, maxelements) {
    override val buffer: ByteBuffer

    init {

        buffer = ByteBuffer.allocateDirect(
                vbdefinition.dataSize * vbdefinition.mSize * maxelements)
                .order(ByteOrder.nativeOrder())
    }

    override fun put(index: Int, value: Byte) {
        buffer.put(index, value)
    }

    override fun put(data: ByteArray, offset: Int) {
        buffer.put(data, offset, data.size - offset)
        buffer.position(0)
    }

    override fun put(data: ByteArray) {
        buffer.put(data, 0, data.size)
        buffer.position(0)
    }

    override fun put(value: Byte) {
        buffer.put(value)
    }

    override fun put(value: Short) {
        buffer.put(value.toByte())
    }

    override fun put(value: Int) {
        buffer.put(value.toByte())
    }

    override fun put(value: Float) {
        buffer.put(value.toByte())
    }

    override fun position(i: Int) {
        buffer.position(i)
    }

    override fun position(): Int {
        return buffer.position()
    }

}